### PR TITLE
ddl: fix inaccurate row_count for `admin show ddl jobs`

### LINKED ISSUE #25968: add index row count doesn't match the actual row count
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
create table t (id bigint key,b int);
split table t by (10),(20),(30);
 insert into t values (0,0),(10,10),(20,20),(30,30);
alter table t add index idx1(b);
admin show ddl jobs;
select count(*) from t;
```

image

### 2. What did you expect to see? (Required)

add index row count match the actual row count.

### 3. What did you see instead (Required)

doesn't match.

### 4. What is your TiDB version? (Required)

```sql
test> select tidb_version()\G
***************************[ 1. row ]***************************
tidb_version() | Release Version: v5.0.2
Edition: Community
Git Commit Hash: dc40a093a0058b95fc859f60d4b84337dbdff9f6
Git Branch: HEAD
UTC Build Time: 2021-07-05 08:12:38
GoVersion: go1.16
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #25968

Problem Summary: you could get information from https://github.com/pingcap/tidb/issues/25968#issuecomment-883949454

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix inaccuate row_count num
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: ddl/backfilling.go [hunk 1]
```diff
@@ -185,13 +185,17 @@ type reorgBackfillTask struct {
...
 
 func (r *reorgBackfillTask) String() string {
 	physicalID := strconv.FormatInt(r.physicalTableID, 10)
 	startKey := tryDecodeToHandleString(r.startKey)
 	endKey := tryDecodeToHandleString(r.endKey)
-	return "physicalTableID_" + physicalID + "_" + "[" + startKey + "," + endKey + "]"
+	if r.endInclude {
+		return "physicalTableID_" + physicalID + "_" + "[" + startKey + "," + endKey + "]"
+	}
+	return "physicalTableID_" + physicalID + "_" + "[" + startKey + "," + endKey + ")"
```
Comment (zimulala) [inline ddl/backfilling.go:198]:
```suggestion
	rangeStr := "physicalTableID_" + physicalID + "_" + "[" + startKey + "," + endKey 
	if r.endInclude {
		return rangeStr + "]"
	}
	return rangeStr + ")"
```

File: ddl/backfilling.go [hunk 2]
```diff
@@ -486,7 +490,9 @@ func (w *worker) sendRangeTaskToWorkers(t table.Table, workers []*backfillWorker
 		task := &reorgBackfillTask{
 			physicalTableID: physicalTableID,
 			startKey:        keyRange.StartKey,
-			endKey:          endKey}
+			endKey:          endKey,
+			// If the boundaries overlap, we should ignore the preceding endKey
```
Comment (zimulala) [inline ddl/backfilling.go:494]:
```suggestion
			// If the boundaries overlap, we should ignore the preceding endKey.
```

**Comment by Defined2014** [comment]
Done, @zimulala PTAL

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**